### DISCUSSION TITLE
Do the recommended Kata Scale with your rank?

### DISCUSSION THREAD

**Discussion by Connor56** [codewars/codewars.com#2945] [General]
I've completed a number of 8kyu Kata and I'm now rank 7 moving up to rank 6 soon.

Does the "next Kata" scale with your rank? Or do you have to deliberately select higher-value Kata?

**Comment by Connor56** [comment]
Surely, this is an easy question to answer?

**Comment by Kacarott** [comment]
I believe they do, however the specifics is a bit unclear. Many people who use Codewars a lot don't even look at the "Next Kata" section, they go directly to the [list of kata](https://www.codewars.com/kata/latest/my-languages?beta=false) and pick one that they want to solve.

**Reply by Connor56** [reply]
Thank you for taking the time to answer!

Why do people do that? What's the thinking behind that process?

**Reply by Kacarott** [reply]
Well, then you get to choose what you want to solve. There's thousands of kata to solve so it makes sense to pick only the ones that look interesting to you.

It's like being able to go to the supermarket to pick whatever food you want vs. someone else making your lunch for you.

**Reply by hobovsky** [reply]
It can also depend on setting of "training focus"

![image](https://user-images.githubusercontent.com/23709795/220890730-69f36095-d485-42e0-8fa7-6359948f366f.png)

**Reply by Connor56** [reply]
@hobovsky Thank you! Unbelievable that I didn't even notice that.

@Kacarott Okay, that makes me think people would accidentally choose Kata that didn't broaden their skill set, but perhaps I'm wrong!

**Reply by Kacarott** [reply]
You are probably right, but keep in mind that many people use Codewars just for fun, and not just for broadening their skills. And if someone wants to broaden their skills it is quite easy to filter for whatever kind of kata you want.

**Reply by hobovsky** [reply]
> Okay, that makes me think people would accidentally choose Kata that didn't broaden their skill set, but perhaps I'm wrong!

Uh how would one _intentionally_ choose a kata to be sure that it broadens one's skill set?  Let alone how would an automated system do this?

**Reply by Connor56** [reply]
@hobovsky By selecting Kata that appear to be different from anything you've seen.

An automated system could do an even better job. Particularly if the people who created the Kata mentioned the expected methods/ tools required to complete the job. i.e. in C, pointers, arrays, or what have you.
